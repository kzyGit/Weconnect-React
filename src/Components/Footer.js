import React, { Component } from 'react';
import '../Styles/App.css';
import { Link } from 'react-router';

/**
 * Display footer, contact information on hover
 */
class Footer extends Component {
  render() {
    let sty = { color: '#eee', textDecoration: 'none' };
    return (
      <div className="footer" ><Link title="+254 701922120, kezzyangiro@gmail.com" style={sty}><span className="glyphicon glyphicon-question-sign"></span> Contact us</Link>
      </div>
    );
  }
}
export default Footer;