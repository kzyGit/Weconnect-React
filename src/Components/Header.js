import React, { Component } from 'react';
import '../Styles/App.css';
import { Link } from 'react-router';
import { browserHistory } from 'react-router';
export const Base_url = process.env.REACT_APP_api_url;

/**
 * Header component
 * When user is logged in, display logout and user tabs
 * When user is not logged in, display login and signup tabs
 */
class Header extends Component {
  render() {

    /**
     * Logout logged in user.
     * Clear access token and loggedIn status from local storage
     */
    this.logout = () => {
      localStorage.removeItem('loggedIn');
      localStorage.removeItem("access_token")
      browserHistory.push('/home')
    }

    let sty = { color: '#eee' }
    let sty2 = { color: '#eee', backgroundColor: 'rgb(54, 54, 129' }
    let droplink = { color: 'black' }
    let sty1 = { borderBottom: '1px solid #ddd', borderRadius: '5px' }
    return (
      <div className="Header" >
        <nav className="navbar navbar-inverse" style={{ backgroundColor: 'rgb(54, 54, 129)', border: '0' }}>
          <div className="container-fluid" >
            <div className="navbar-header">

              <button type="button" className="navbar-toggle" data-toggle="collapse" data-target="#myNavbar">
                <span className="icon-bar"></span>
                <span className="icon-bar"></span>
                <span className="icon-bar"></span>
              </button>

              <Link to="/home" className="navbar-brand" style={{ color: '#eee', fontSize: '25px', marginLeft: '1px' }} >WeConnect *</Link>
            </div>
            <div className="collapse navbar-collapse" id="myNavbar" >

              <ul className="nav navbar-nav navbar-right">
                <li><Link to="/home" style={sty} activeStyle={sty1}><span className="glyphicon glyphicon-home"></span> Home</Link></li>
                <li><Link to="/businesses" style={sty} activeStyle={sty1}>Businesses</Link></li>

                {localStorage.loggedIn &&
                  <li>
                    <a style={sty2} data-toggle="dropdown"><span className="glyphicon glyphicon-user"></span><span id='logoutname'>{localStorage.getItem("username")} </span>
                      <span className="caret" ></span></a>
                    <ul className="dropdown-menu" style={{ backgroundColor: '#ddd' }}>
                      <li><Link to="/addbusiness" style={droplink}><span style={{ paddingRight: '10px' }} className="glyphicon glyphicon-plus-sign"></span> Add Business</Link></li>
                      <li><Link to="/dashboard" style={droplink}><span style={{ paddingRight: '10px' }} className="glyphicon glyphicon-cog"></span> Dashboard</Link></li>
                      <li><Link to={'/editPassword'}><span style={{ paddingRight: '15px' }} className="glyphicon glyphicon-edit"></span>Edit password</Link></li>
                      <li><Link onClick={this.logout.bind(this)} ><span style={{ paddingRight: '10px' }} className="glyphicon glyphicon-log-out"></span> Logout</Link></li>
                    </ul>
                  </li>
                }
                {!localStorage.loggedIn && <li><Link to="/login" style={sty} activeStyle={sty1}><span className="glyphicon glyphicon-log-in"></span> Login</Link></li>}
                {!localStorage.loggedIn && <li><Link to="/signup" style={sty} activeStyle={sty1}><span className="glyphicon glyphicon-user"></span> Sign Up</Link></li>}
              </ul>

            </div>
          </div>
        </nav>
      </div>
    );
  }
}

export default Header;