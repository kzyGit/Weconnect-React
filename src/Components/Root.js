import React, { Component } from 'react';
import '../Styles/App.css';
import PropTypes from 'prop-types'

class Root extends Component {
  render() {
    return (
      <div className="row">
        {this.props.children}
      </div>
    );
  }
}
Root.propTypes = {
  children: PropTypes.object.isRequired
}
export default Root;