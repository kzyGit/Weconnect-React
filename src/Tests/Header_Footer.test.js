import React from 'react';
import { shallow } from 'enzyme';
import Footer from '../Components/Footer';
import Header from '../Components/Header';
import Root from '../Components/Root';
import { shallowToJson } from 'enzyme-to-json';


describe('Footer component', () => {
    const wrapper = shallow(<Footer />);

    it('renders properly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});

describe('Root component', () => {
    const wrapper = shallow(<Root />);

    it('renders properly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});

describe('Header component', () => {

    const localStorageMock = {
        getItem: jest.fn(),
        setItem: jest.fn(),
        clear: jest.fn()
    };
    global.localStorage = localStorageMock;

    const params = {
        params: {
            id: 1,
        },
    };

    const wrapper = shallow(<Header params={{ params }} />);

    it('renders properly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});

