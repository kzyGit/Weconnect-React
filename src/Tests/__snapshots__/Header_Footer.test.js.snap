// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders properly 1`] = `
<div
  className="footer"
>
  <Link
    onlyActiveOnIndex={false}
    style={
      Object {
        "color": "#eee",
        "textDecoration": "none",
      }
    }
    title="+254 701922120, kezzyangiro@gmail.com"
  >
    <span
      className="glyphicon glyphicon-question-sign"
    />
     Contact us
  </Link>
</div>
`;

exports[`Header component renders properly 1`] = `
<div
  className="Header"
>
  <nav
    className="navbar navbar-inverse"
    style={
      Object {
        "backgroundColor": "rgb(54, 54, 129)",
        "border": "0",
      }
    }
  >
    <div
      className="container-fluid"
    >
      <div
        className="navbar-header"
      >
        <button
          className="navbar-toggle"
          data-target="#myNavbar"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="icon-bar"
          />
          <span
            className="icon-bar"
          />
          <span
            className="icon-bar"
          />
        </button>
        <Link
          className="navbar-brand"
          onlyActiveOnIndex={false}
          style={
            Object {
              "color": "#eee",
              "fontSize": "25px",
              "marginLeft": "1px",
            }
          }
          to="/home"
        >
          WeConnect *
        </Link>
      </div>
      <div
        className="collapse navbar-collapse"
        id="myNavbar"
      >
        <ul
          className="nav navbar-nav navbar-right"
        >
          <li>
            <Link
              activeStyle={
                Object {
                  "borderBottom": "1px solid #ddd",
                  "borderRadius": "5px",
                }
              }
              onlyActiveOnIndex={false}
              style={
                Object {
                  "color": "#eee",
                }
              }
              to="/home"
            >
              <span
                className="glyphicon glyphicon-home"
              />
               Home
            </Link>
          </li>
          <li>
            <Link
              activeStyle={
                Object {
                  "borderBottom": "1px solid #ddd",
                  "borderRadius": "5px",
                }
              }
              onlyActiveOnIndex={false}
              style={
                Object {
                  "color": "#eee",
                }
              }
              to="/businesses"
            >
              Businesses
            </Link>
          </li>
          <li>
            <Link
              activeStyle={
                Object {
                  "borderBottom": "1px solid #ddd",
                  "borderRadius": "5px",
                }
              }
              onlyActiveOnIndex={false}
              style={
                Object {
                  "color": "#eee",
                }
              }
              to="/login"
            >
              <span
                className="glyphicon glyphicon-log-in"
              />
               Login
            </Link>
          </li>
          <li>
            <Link
              activeStyle={
                Object {
                  "borderBottom": "1px solid #ddd",
                  "borderRadius": "5px",
                }
              }
              onlyActiveOnIndex={false}
              style={
                Object {
                  "color": "#eee",
                }
              }
              to="/signup"
            >
              <span
                className="glyphicon glyphicon-user"
              />
               Sign Up
            </Link>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;

exports[`Root component renders properly 1`] = `
<div
  className="row"
>
  <Header />
  <BackNav />
  <Footer />
</div>
`;
